PLC Development/DrivePLC/POUs/FB_AxisDrive.TcPOU

```diff
@@ -0,0 +1,868 @@
+		MCMoveAbsolute.Acceleration:=AxisController.fAcceleration;
+		MCMoveAbsolute.Deceleration:=AxisController.fDeceleration;
+		MCMoveAbsolute.Jerk:=0;
+		MCMoveAbsolute.BufferMode:=MC_BufferMode.MC_Buffered;
+		
+		bDone:=MCMoveAbsolute.Done;
+		AxisController.bBusy:=MCMoveAbsolute.Busy;
+		
+		IF MCMoveAbsolute.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCMoveAbsolute.ErrorID;
+		END_IF
+		IF NOT MCMoveAbsolute.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.MoveRelative:		
+		MCMoveRelative.Execute:=AxisController.bExecute;
+		MCMoveRelative.Distance:=AxisController.fDistance;
+		MCMoveRelative.Velocity:=AxisController.fVelocity;
+		MCMoveRelative.Acceleration:=AxisController.fAcceleration;
+		MCMoveRelative.Deceleration:=AxisController.fDeceleration;
+		MCMoveRelative.Jerk:=0;
+		MCMoveRelative.BufferMode:=MC_BufferMode.MC_Buffered;
+		
+		bDone:=MCMoveRelative.Done;
+		AxisController.bBusy:=MCMoveRelative.Busy;
+		
+		IF MCMoveRelative.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCMoveRelative.ErrorID;
+		END_IF
+		IF NOT MCMoveRelative.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF		
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.GearIn:
+		//Dependent on +ve or -ve gear ratio determines whether to link Fw to Fw and Bw to Bw or Fw to Bw and Bw to Fw
+		IF AxisController.fGearRatioNum>0 THEN
+			Limits.coupledFwLimit[GVL.iAxis][1].eLimitType:=eLimitCoupling.FwSw;
+			Limits.coupledFwLimit[GVL.iAxis][1].iMasterAxis:=AxisController.masterAxis;
+			Limits.coupledBwLimit[GVL.iAxis][1].eLimitType:=eLimitCoupling.BwSw;
+			Limits.coupledBwLimit[GVL.iAxis][1].iMasterAxis:=AxisController.masterAxis;
+			
+		END_IF
+		IF AxisController.fGearRatioNum<0 THEN
+			Limits.coupledFwLimit[GVL.iAxis][1].eLimitType:=eLimitCoupling.BwSw;
+			Limits.coupledFwLimit[GVL.iAxis][1].iMasterAxis:=AxisController.masterAxis;
+			Limits.coupledBwLimit[GVL.iAxis][1].eLimitType:=eLimitCoupling.FwSw;
+			Limits.coupledBwLimit[GVL.iAxis][1].iMasterAxis:=AxisController.masterAxis;
+		END_IF
+		
+		MCGearIn.Execute:=AxisController.bExecute;
+		MCGearIn.RatioNumerator:=AxisController.fGearRatioNum;
+		MCGearIn.RatioDenominator:=AxisController.iGearRatioDen;
+		
+		bDone:=McGearIn.InGear;
+		AxisController.bBusy:=MCGearIn.Busy;
+		
+		IF MCGearIn.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCGearIn.ErrorID;
+		END_IF		
+		IF NOT MCGearIn.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.GearInMultiMaster:		
+		//Set up limits
+		FOR iMasterIndex:=1 TO 4 BY 1 DO
```

Comment:
Would be good to have a constant somewhere that is the number of axes you can couple.